scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,197 @@
+            vectorizers=[
+                AzureOpenAIVectorizer(
+                    name=f"{search_info.index_name}-vectorizer",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=AzureOpenAIParameters(
+                        resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+                        deployment_id=self.embeddings.open_ai_deployment,
+                    ),
+                ),
+            ]
+        )
+
+        # create indexer client
+        ds_client = search_info.create_search_indexer_client()
+        ds_container = SearchIndexerDataContainer(name=self.blob_manager.container)
+        data_source_connection = SearchIndexerDataSourceConnection(
+            name=f"{search_info.index_name}-blob",
+            type="azureblob",
+            connection_string=self.blob_manager.get_managedidentity_connectionstring(),
+            container=ds_container,
+            data_deletion_detection_policy=NativeBlobSoftDeleteDeletionDetectionPolicy(),
+        )
+
+        await ds_client.create_or_update_data_source_connection(data_source_connection)
+        print("Search indexer data source connection updated.")
+
+        embedding_skillset = await self.create_embedding_skill(search_info.index_name)
+        await ds_client.create_or_update_skillset(embedding_skillset)
+
+    async def run(self, search_info: SearchInfo):
+        if self.document_action == DocumentAction.Add:
+            files = self.list_file_strategy.list()
+            async for file in files:
+                try:
+                    await self.blob_manager.upload_blob(file)
+                finally:
+                    if file:
+                        file.close()
+        elif self.document_action == DocumentAction.Remove:
+            paths = self.list_file_strategy.list_paths()
+            async for path in paths:
+                await self.blob_manager.remove_blob(path)
+        elif self.document_action == DocumentAction.RemoveAll:
+            await self.blob_manager.remove_blob()
+
+        # Create an indexer
+        indexer_name = f"{search_info.index_name}-indexer"
+
+        indexer = SearchIndexer(
+            name=indexer_name,
+            description="Indexer to index documents and generate embeddings",
+            skillset_name=f"{search_info.index_name}-skillset",
+            target_index_name=search_info.index_name,
+            data_source_name=f"{search_info.index_name}-blob",
+            # Map the metadata_storage_name field to the title field in the index to display the PDF title in the search results
+            field_mappings=[FieldMapping(source_field_name="metadata_storage_name", target_field_name="title")],
+        )
+
+        indexer_client = search_info.create_search_indexer_client()
+        indexer_result = await indexer_client.create_or_update_indexer(indexer)
+
+        # Run the indexer
+        await indexer_client.run_indexer(indexer_name)
```

Comment:
this crashes if another indexer run is in progress

azure.core.exceptions.ResourceExistsError: () Another indexer invocation is currently in progress; concurrent invocations are not allowed.
Code:
Message: Another indexer invocation is currently in progress; concurrent invocations are not allowed.

Reply:
Created follow up item